features/step_definitions/sd-step.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+/* eslint-disable no-unused-vars */
+
+const Assert = require('chai').assert;
+const request = require('../support/request');
+const sdapi = require('../support/sdapi');
+const TIMEOUT = 240 * 1000;
+
+module.exports = function server() {
+    // eslint-disable-next-line new-cap
+    this.Before({
+        tags: ['@sd-step']
+    }, () => {
+        this.repoOrg = this.testOrg;
+        this.repoName = 'functional-shared-steps';
+        this.pipelineId = null;
+        this.eventId = null;
+        this.meta = null;
+        this.jwt = null;
+        this.image = null;
+        this.expectedImage = null;
+        this.expectedPackage = null;
+        this.commands = null;
+    });
+
+    this.Given(/^an existing pipeline with these images and packages with version:$/,
+        { timeout: TIMEOUT }, table =>
+        this.getJwt(this.accessKey)
+        .then((response) => {
+            this.jwt = response.body.token;
+            this.expectedImage = table.rows()[0][0];
+            this.expectedPackage = table.rows()[0][1];
+
+            return request({
```

Comment:
I agree. This `POST` request is used by other step definitions.
And I think `checkoutUrl` should be configurable (SCM url and branch).